File: 0513-Find-Bottom-Left-Tree-Value/Java/Solution.java
import java.util.LinkedList;
import java.util.Queue;

public class Solution {
    
    /**
     * https://leetcode.cn/problems/find-bottom-left-tree-value/
     * 
     * @param root
     * @return
     */
    public int findBottomLeftValue(TreeNode root) {
        int leftValue = root.val;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            TreeNode node = queue.poll();
            if (node.right != null) {
                leftValue = node.right.val;
                queue.add(node.right);
            }
            if (node.left != null) {
                leftValue = node.left.val;
                queue.add(node.left);
            }
        }
        return leftValue;
    }
}

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}   

File: 0513-Find-Bottom-Left-Tree-Value/Python/Solution.py
from typing import Optional, Deque
from collections import deque


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        result: int = -1

        q: Deque[TreeNode] = deque()
        q.append(root)

        while q:
            size: int = len(q)
            if size:
                result = q[0].val

            for _i in range(size):
                tn = q.popleft()
                if tn.left:
                    q.append(tn.left)
                if tn.right:
                    q.append(tn.right)

        return result
